transport_plugins/bled112/scripts/profile_gateway.py

```diff
@@ -0,0 +1,63 @@
+"""
+Runs gateway instance and yappi in order to profile it,
+also opens HWManager websocket to the gateway
+"""
+
+import time
+import logging
+import argparse
+import yappi
+from iotile.core.hw import HardwareManager
+from iotile.core.utilities import BackgroundEventLoop
+from iotilegateway.gateway import IOTileGateway
+
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+
+gateway_config = {
+    "adapters": [
+        {
+            "name": "bled112",
+            "port": "/dev/pts/4"
+        }
+    ],
+    "servers": [
+        {
+            "name": "websockets",
+            "args":
+            {
+                "port": 5120
+            }
+        }
+    ]
+}
+
+def profile_gateway(config, file_name="log.txt", time_to_profile=300):
+
+    loop = BackgroundEventLoop()
+    loop.start()
+
+    yappi.start()
+
+    gateway = IOTileGateway(config, loop)
+    loop.run_coroutine(gateway.start())
+
+    hw = HardwareManager(port="ws:127.0.0.1:5120/iotile/v1")
+
+    time.sleep(time_to_profile)
```

Comment:
This should be wrapped in a `try/except KeyboardInterrupt: pass` so that if you control C you can exit early cleanly.  As written it hangs the program because the main thread exits but it never cleans up the background event loop so python hangs at exit doing a `join` on the background thread.

Reply:
I added try/except clauses and loop.stop() call